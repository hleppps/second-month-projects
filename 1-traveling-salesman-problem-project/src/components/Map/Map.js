import React from "react";
import { ComposableMap, Geographies } from "react-simple-maps";
import Lines from "../Lines";
import Markers from "../Markers";
import styles from "./Map.module.css";

const Map = React.memo((props) => {
  return (
    <ComposableMap
      className={styles.map}
      projectionConfig={{
        center: [9, 49],
        scale: 1100,
      }}
    >
      <Geographies data-testid="geographies" geography={props.geoUrl}>
        {props.countriesDrawHandler}
      </Geographies>
      <Lines />
      <Markers />
    </ComposableMap>
  );
});

export default Map;
