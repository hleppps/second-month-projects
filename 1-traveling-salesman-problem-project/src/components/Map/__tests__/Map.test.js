import { cleanup, render } from "@testing-library/react";
import React from "react";
import Map from "../Map";

afterEach(cleanup);

it("matches a snapshot", () => {
  const { asFragment } = render(<Map />);
  expect(asFragment(<Map />)).toMatchSnapshot();
});
