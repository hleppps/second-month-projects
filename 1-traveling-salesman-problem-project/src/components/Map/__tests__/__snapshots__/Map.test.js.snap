// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<DocumentFragment>
  <svg
    class="rsm-svg map"
    viewBox="0 0 800 600"
  >
    <g
      class="rsm-geographies "
      data-testid="geographies"
    />
  </svg>
</DocumentFragment>
`;
